tests/zfs-tests/tests/functional/redacted_send/redacted_deleted.ksh

```diff
@@ -96,7 +96,7 @@ log_must zfs destroy -R $clone2
 log_must eval "zfs send -i $sendfs#book2 --redact book3 $sendfs@snap2 >$stream"
 log_must eval "zfs recv $recvfs <$stream"
 log_must mount_redacted -f $recvfs
-log_must diff <(ls $send_mnt) <(ls $recv_mnt)
+log_must [ "$(ls $send_mnt)" == "$(ls $recv_mnt)" ]
```

Comment:
The `directory_diff` helper function may be a better fit here.  See `zfs-tests/include/libtest.shlib`.

Reply:
Weirdly `directory_diff` is failing and printing:
```
>fc........ f2
```
Maybe it's doing something more complicated than just a `ls` comparison?

Reply:
Ok, the test case directories being compared have a file in both of them that appears to be the same:
```
$ ls -la /testpool/deleted/ /testpool2/deleted/
/testpool2/deleted/:
total 2
drwxr-xr-x. 2 root root       3 May 29 15:01 .
drwxr-xr-x. 3 root root       3 May 29 15:01 ..
-rw-r--r--. 1 root root 4194304 May 29 15:01 f2

/testpool/deleted/:
total 4102
drwxr-xr-x. 2 root root       3 May 29 15:01 .
drwxr-xr-x. 5 root root       5 May 29 15:01 ..
-rw-r--r--. 1 root root 4194304 May 29 15:01 f2
```
However, the files themselves are different:
```
$ md5sum /testpool/deleted/f2
5df78ee14021a16df23fedd038efba26  /testpool/deleted/f2

$ md5sum /testpool2/deleted/f2
b5cfa9d6c8febd618f91ac2843d50a1c  /testpool2/deleted/f2
```
I assume that's ok, since the original test code was comparing the `ls` output rather than the files themselves?

Reply:
Yes, I believe that's right since the redacted records shouldn't be included in the stream.

Reply:
Ok, so sounds like current `[ "$(ls $send_mnt)" == "$(ls $recv_mnt)" ]` comparison I have is ok then.